### PR TITLE
fix: allow testing any issuer

### PR DESCRIPTION
**Pull request by brownoxford**
## What kind of change does this PR introduce?

This PR updates the JWT verification code so that if the user has specified multiple allowed issuers, the verification works properly.

Fixes: #32

### CODE REVIEW

**Comment by chrisl8** [comment]
I tested this and it works as advertised and also does not cause any issues with existing usage.

**Review by fenix-hub — COMMENTED** [review]
`expected_claims.get(claim)` in this case should always return a PackedStringArray, since the [`with_any_issuers`](https://github.com/brownoxford/godot-engine.jwt/blob/09bc30e2547ccf1d04c0671fef2ff91d8fe272d3/addons/jwt/src/JWTVerifierBuilder.gd#L31) methods adds a PackedStringArray as a claim.
Wouldn't it make more sense to just consider changes from line 72 + ?

```
if not expected_claims.get(claim).has(jwt_decoder.get_issuer()):
  ...
```

@chrisl8 @brownoxford

**Comment by brownoxford** [comment]
@fenix-hub I'm not sure we can do that here, since that would mean that _all_ claims are `PackedStringArray`. The purpose of this change is to account for the specific `JWTClaims.Public.ISSUER` claim being _either_ a `String` _or_ a `PackedStringArray`, since `JWTBaseBuilder` has this:

```
func with_issuer(issuer: String) -> JWTBaseBuilder:
	add_claim(JWTClaims.Public.ISSUER, issuer)
	return self
```

and `JWTVerifierBuilder` has this:

```
func with_any_of_issuers(issuers: PackedStringArray) -> JWTVerifierBuilder:
	add_claim(JWTClaims.Public.ISSUER, issuers)
	return self
```

In truth, verifying against multiple possible issuers seems like an edge or maybe even corner case to me, and should probably be pushed off to the application instead of being in the implementation. What I mean by that is that our verification code in this implementation should verify an issuer string against an issuer claim string when needed, otherwise, it would be up to the application using this implementation to check against multiple possible allowed values.

I think that is an issue for another time though, and is relevant to the discussion we recently started.

**Comment by chrisl8** [comment]
I admit I didn't really understand **why** this code was here, I just tested that it worked with both a single and multiple issuers (String or Array) and that it didn't break existing functionality.

I think I agree with that this feature might not even belong in the library. If someone has an unusual requirement (allow one of many issuers) then they can just **not** validate the issuer and then check it in the decoded JWT output after it is verified.

In theory someone will be putting any number of "payloads" or "claims" (as far as I know they are the same in the token itself) into the JWT, and will need to do their own checking post-verification that they exist.

I noticed that the Python JWT library mentioned elsewhere has an option to also verify that a given "claim" **exists** regardless of what the content is. This one on the other hand only lets me check a claim with a specific value. How would I know, for instance, what user ID to check for before decoding the JWT?
However, again, I'm not sure it is highly valuable for a library to let me make arbitrary checks on the contents when I am going to have to process the contents later anyway.
So even the Python JWT library probably has more functions that it needs to in my opinion.

For the JWT spec I believe that there are "registered claims", like the issuer, expiration, etc. and it makes sense to include those as part of the library, but everything else is probably an exercise for code written and run against the decoded value later by the developer, not this library.

TL;DR: I agree that this is probably a feature that we don't even need, but since it was apparently already here, it is good for it to work. I would reverse my position if the RFC indicates that the issuer "registered claim" is supposed to allow one or many entries.
